tests/run/pyintop.pyx

```diff
@@ -515,3 +515,130 @@ def test_avoid_if_coercion(obj):
         return True
     else:
         return False
+
+def pure_add_int(obj2: int):
```

Comment:
Probably worth adding `test_fail_if_path_exists`/`test_assert_path_exists`. I'd guess it should fail with `BinopNode`, and we should require a `CApiCallNode`. But I'm not completely sure of this off the top of my head.

For debugging purposes it's useful to temporarily add `PrintTree()` to the end of the pipeline in `Pipeline.py` - I think there might be a command line argument to do it but I find it easier just to edit the code

Reply:
> Probably worth adding test_fail_if_path_exists/test_assert_path_exists. I'd guess it should fail with BinopNode, and we should require a CApiCallNode

I was thinking about that. I did not added them because the PR is not changing tree but I will try to add them.

Reply:
> For debugging purposes it's useful to temporarily add PrintTree()

I suppose this is just a note, It should not be added in this PR...

Reply:
I believe the PR will change the tree - it replaces a `BinopNode` with a function call of the optimized function.

> I  suppose this is just a note, It should not be added in this PR...

Yes - it was just a hint about how to work out what the tree is changed to.

Reply:
> I believe the PR will change the tree
Hmm OK I was not aware of that. I am still stranger in Cython internals...

Reply:
OK done. It was good hint because I found out that PR was not exercised because annotations was ignored. Hence, there are two major changes:
1. language_level was set to 3, I hope it does not affect rest of the tests
2. I have removed tests for non-Commutativity operations because it generates different code when operators are reversed.